sqlglot/generator.py

```diff
@@ -2399,18 +2399,21 @@ def ordered_sql(self, expression: exp.Ordered) -> str:
                     f"'{nulls_sort_change.strip()}' translation not supported in window functions"
                 )
                 nulls_sort_change = ""
-            elif (
-                self.NULL_ORDERING_SUPPORTED is False
-                and (isinstance(expression.find_ancestor(exp.AggFunc, exp.Select), exp.AggFunc))
-                and (
-                    (asc and nulls_sort_change == " NULLS LAST")
-                    or (desc and nulls_sort_change == " NULLS FIRST")
-                )
+            elif self.NULL_ORDERING_SUPPORTED is False and (
+                (asc and nulls_sort_change == " NULLS LAST")
+                or (desc and nulls_sort_change == " NULLS FIRST")
             ):
-                self.unsupported(
-                    f"'{nulls_sort_change.strip()}' translation not supported for aggregate functions with {sort_order} sort order"
-                )
-                nulls_sort_change = ""
+                # BigQuery does not allow these ordering/nulls combinations when used under
+                # an aggregation func or under a window containing one
+                ancestor = expression.find_ancestor(exp.AggFunc, exp.Window, exp.Select)
+
+                if ancestor and any(
+                    isinstance(expr, exp.AggFunc) for expr in (ancestor, ancestor.this)
+                ):
```

Comment:
This is a little difficult to understand, and I think there may be an issue if `ancestor` ends up being a `Select`, because you don't check for that anymore. I'd suggest making it more explicit in favor of readability:

```python
if isinstance(ancestor, exp.Window):
    ancestor = ancestor.this
if isinstance(ancestor, exp.AggFunc):
    self.unsupported(...)
    nulls_sort_change = ""
```